### ISSUE TITLE
smarty and attr_list conflict

### ISSUE THREAD

**Issue by facelessuser** [Python-Markdown/markdown#324]
This:

```
[link](http://example.com){: class="foo bar" title="Some title!" }
```

Produces this:

``` html
<a bar_wzxhzdk:0_="bar”" class="”foo" href="http://example.com" title="”Some" title_wzxhzdk:1_="title!”">link</a>
```

Current workaround is to disable `smart_quotes`:

```
"smarty(smart_quotes=False)",
```

I took a look, and it looks to be a fairly involved problem to fix if `attr_list` was to keep the same rigid syntax requirement since `smarty` appears to occur as an `inline` before `attr_list` in `treeprocessors`.  I suppose that `attr_list` could potentially allow for fancy quotes in its regex.  I believe that would be the simplest fix.

I wouldn't mind submitting a pull, but I would need an okay on this direction before I proceed.

**Comment by mitya57** [comment]
This is already fixed in Git (see #311).

**Comment by facelessuser** [comment]
Great! Never mind the.
